web/components/App.jsx

```diff
@@ -0,0 +1,215 @@
+/**
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/* eslint-disable no-console */
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import moment from 'moment';
+import _ from 'lodash';
+import FilterHeader from './FilterHeader.jsx';
+import EventMessage from './EventMessage.jsx';
+import UserMessage from './UserMessage.jsx';
+import ChatMessage from './ChatMessage.jsx';
+import AttachmentMessage from './AttachmentMessage.jsx';
+import ToastMessage from './ToastMessage.jsx';
+import ChatBox from './ChatBox.jsx';
+import './chat.css';
+
+const { env } = require('../../config.js');
+const config = require('../../config.js')[env];
+const botName = require('../../package.json').name;
+const bdk = require('@salesforce/refocus-bdk')(config, botName);
+import MessageService from '../services/MessageService';
+import AttachmentService from '../services/AttachmentService';
+
+const ZERO = 0;
+const DFB = 5; // Distance From Bottom
+const messageService = new MessageService(bdk);
+const attachmentService = new AttachmentService(bdk);
+
+/**
+ * @param {object} props
+ * @returns {JSX} app container.
+ */
+export default function App(props) {
```

Comment:
I haven't used hooks yet, but I'm guessing that's the reason you export a "function" and not a "class" App here. Of course classes are just functions, but I didn't realise having function definitions inside functions was a thing :P

Reply:
Exactly, it eliminates the need for `constructor`, `componentDidMount` etc. You can just define things like `const [toast, setToast] = useState(false);` & manage state that way